temba/msgs/models.py

```diff
@@ -789,7 +791,15 @@ def send_messages(cls, all_msgs):
         if task_msgs:
             if task_priority is None:
                 task_priority = DEFAULT_PRIORITY
-            push_task(task_msgs[0]['org'], MSG_QUEUE, SEND_MSG_TASK, task_msgs, priority=task_priority)
+
+            tasks_to_push.append((task_msgs, task_priority))
+
+        on_transaction_commit(lambda: cls.push_tasks(tasks_to_push))
+
+    @classmethod
+    def push_tasks(cls, tasks):
```

Comment:
I don't think there's any real need for this, it is basically just recreating what `on_transaction_commit` already does. (it has an internal list, so you can call `on_transaction_commit` more than once)

Just changing all the push_task to `on_transaction_commit` should work no? (being careful about shared variables)